### PR TITLE
#2902 - extend event participant jurisdiction calculation

### PR DESCRIPTION
**Pull request by sergiupacurariu**

### CODE REVIEW

**Review by leventegal-she — APPROVED** [review]

File: sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java [hunk 1]
```diff
@@ -193,8 +193,8 @@ public EventParticipantDto saveEventParticipant(EventParticipantDto dto) {
 		Event event = eventService.getByUuid(eventReferenceDto.getUuid());
 
 		if (!eventJurisdictionChecker.isInJurisdiction(event) && (dto.getRegion() == null || dto.getDistrict() == null)) {
-			dto.setRegion(new RegionReferenceDto(user.getRegion().getUuid()));
-			dto.setDistrict(new DistrictReferenceDto(user.getDistrict().getUuid()));
+			dto.setRegion(new RegionReferenceDto(user.getRegion() != null ? user.getRegion().getUuid() : null));
```
Comment (leventegal-she) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java:196]: should be `user.getRegion() ? new RegionReferenceDto() : null`

File: sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java [hunk 2]
```diff
@@ -193,8 +193,8 @@ public EventParticipantDto saveEventParticipant(EventParticipantDto dto) {
...
+			dto.setDistrict(new DistrictReferenceDto(user.getDistrict() != null ? user.getDistrict().getUuid() : null));
```
Comment (leventegal-she) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java:197]: same as above

File: sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventParticipantsController.java [hunk 3]
```diff
@@ -208,15 +210,34 @@ public void deleteEventParticipant(String eventUuid, String personUuid, Runnable
...
+				RegionReferenceDto userRegion = user.getRegion();
+				DistrictReferenceDto userDistrict = user.getDistrict();
+
+				RegionReferenceDto epResponsibleRegion = dto.getRegion();
+				DistrictReferenceDto epResponsibleDistrict = dto.getDistrict();
+
+				RegionReferenceDto epEventRegion = eventDto.getEventLocation().getRegion();
+				DistrictReferenceDto epEventDistrict = eventDto.getEventLocation().getDistrict();
+
+				Boolean responsibleRegionDiffersFromUserRegion =
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventParticipantsController.java:223]: I would chose better naming for the boolean variables

**Review by leventegal-she — CHANGES_REQUESTED** [review]

File: sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java (hunk 1 above)
Comment (sergiupacurariu) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java:196]: changed

File: sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java (hunk 2 above)
Comment (sergiupacurariu) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java:197]: changed

File: sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventParticipantsController.java (hunk 3 above)
Comment (sergiupacurariu) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventParticipantsController.java:224]: I've made some changes but I could not get more inspired than that. I also added comments explaining each boolean variable.

**Review by leventegal-she — APPROVED** [review]

**Pull request merged**